src/feature/hs/hs_client.c

```diff
@@ -1233,7 +1254,7 @@ hs_client_decode_descriptor(const char *desc_str,
   }
 
   /* Parse descriptor */
-  ret = hs_desc_decode_descriptor(desc_str, subcredential, desc);
+  ret = hs_desc_decode_descriptor(desc_str, subcredential, client_sk, desc);
   memwipe(subcredential, 0, sizeof(subcredential));
   if (ret < 0) {
     log_warn(LD_GENERAL, "Could not parse received descriptor as client.");
```

Comment:
Client *without* any client authorization configured but the service descriptor requires authorization. This means, the client won't be able to decrypt the descriptor and thus ending up here.

We should first bring down that log to `log_info()` in my opinion. This is not something the user needs to know at the warning level, nothing to be concerned about I would say.

Second, and we could do that after the merge, would be to make the error code from `hs_desc_decode_descriptor()` a bit more useful like "Ok descriptor wasn't decryptable" which could make the client print a log message like "We probably do not have access to A.onion. Enable client authorization" kind of thing. And *maybe*, we could also stop the client from refetching a new descriptor from a new HSDir...